docs/xenit-style-guide/golang.md

```diff
@@ -7,6 +7,65 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
 Golang (Go) is one of the most common languages at Xenit especially for backend systems and open source projects. It should be the first language choice when starting a new project.
 
+## Startup and Shutdown
+
+It may seem a bit nit picky to document how a program should startup and shutdown, but it is necissary as there are a lot of resources like blogs which offer a multitude of solutions how it could be implemented. Doing this properly is important to make sure that all incoming messages such as HTTP requests and processed before shutting down, the alternative would be to cancel HTTP request without responding to them. There are generally two event soures which cause a shutdown, either an internal error which requires the program shutdown or an [external signal](https://pkg.go.dev/os/signal) notifying the program to shut down.
+
+In a lot of cases a program may run multiple go routines simultaneously, and example of this would be running both the buisness logic HTTP server and the HTTP server that is serving metrics. All of these would need to be gracefully stopped and also cause a graceful shutdown in case of an error. The [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup) package offers a solution to this problem by wrapping the go routines with an error handler which cancels a context when one of the go routines returns an error.
+
+```golang
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
```

Comment:
Should we recommend having a `main()` as well as a `run() error`?

Reply:
I think it is better to not do that right now.